Generators/src/GeneratorFromFile.cxx

```diff
@@ -253,6 +278,13 @@ bool GeneratorFromO2Kine::importParticles()
       auto pdg = t.GetPdgCode();
       auto px = t.Px();
       auto py = t.Py();
+      if (mRandomPhi) {
+        auto phi = TMath::ATan2(py, px);
+        auto pt = TMath::Sqrt(px * px + py * py);
+        phi += dPhi;
+        px = pt * TMath::Cos(phi);
+        py = pt * TMath::Sin(phi);
+      }
```

Comment:
I believe this can be done with less cost by applying the rotation as a matrix - vector multiplication:

```
px =  cos phi * px_old + sin (-phi) * py_old 
py =  sin phi * px_old + cos (phi) * py_old
```
This would only need evaluation of cos, sin and not Atan2 nor sqrt.